File: 1926-nearest-exit-from-entrance-in-maze/1926-nearest-exit-from-entrance-in-maze.py
class Solution:
    def nearestExit(self, maze: List[List[str]], entrance: List[int]) -> int:
        m = len(maze)
        n = len(maze[0])
        que = deque([entrance])
        chk = [[-1 for i in range(n)] for j in range(m)]
        chk[entrance[0]][entrance[1]] = 0

        dr = [0, 0, -1, 1]
        dc = [1, -1, 0, 0]

        while que:
            r, c = que.popleft()
            if ((r == 0 or c == 0 or r == m - 1 or c == n - 1) and
                not (r == entrance[0] and c == entrance[1])):
                return chk[r][c]

            for i in range(4):
                nr, nc = r + dr[i], c + dc[i]
                if (0 <= nr < m and 0 <= nc < n and
                    chk[nr][nc] == -1 and
                    maze[nr][nc] == '.'):
                    chk[nr][nc] = chk[r][c] + 1
                    que.append((nr, nc))

        return -1

File: 1926-nearest-exit-from-entrance-in-maze/1926-nearest-exit-from-entrance-in-maze.java
class Coordinate {
    int r;
    int c;

    public Coordinate(int r, int c) {
        this.r = r;
        this.c = c;
    }
}

class Solution {
    public int nearestExit(char[][] maze, int[] entrance) {
        int m = maze.length;
        int n = maze[0].length;
        Queue<Coordinate> que = new ArrayDeque<>();
        que.add(new Coordinate(entrance[0], entrance[1]));
        int[][] chk = new int[m][n];
        chk[entrance[0]][entrance[1]] = 1;

        int[] dr = {0, 0, -1, 1};
        int[] dc = {1, -1, 0, 0};

        while (!que.isEmpty()) {
            Coordinate now = que.poll();
            int r = now.r;
            int c = now.c;

            if ((r == 0 || c == 0 || r == m - 1 || c == n - 1) &&
                !(r == entrance[0] && c == entrance[1])) {
                return chk[r][c] - 1;
            }

            for (int i = 0; i < 4; i++) {
                int nr = r + dr[i];
                int nc = c + dc[i];
                if (0 <= nr && nr < m && 0 <= nc && nc < n &&
                    chk[nr][nc] == 0 && maze[nr][nc] == '.') {
                    chk[nr][nc] = chk[r][c] + 1;
                    que.add(new Coordinate(nr, nc));
                }
            }
        }

        return -1;
    }
}